charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -38,6 +38,8 @@ spec:
         "vault.hashicorp.com/agent-inject-template-serverca.crt": {{ template "consul.serverTLSCATemplate" . }}
         {{- if .Values.global.secretsBackend.vault.agentAnnotations }}
         {{ tpl .Values.global.secretsBackend.vault.agentAnnotations . | nindent 8 | trim }}
+        {{- else if .Values.global.secretsBackend.vault.vaultNamespace }}
```

Comment:
Here and in other files:

Should this be an `else`? I might be wrong but I assumed you could specify `agentAnnotations` as well as `vaultNamespace` at the same time (annotations can include [many](https://developer.hashicorp.com/vault/docs/platform/k8s/injector/annotations#agent-annotations) values, not just namespace).

Seems this might also be lacking test coverage - maybe something we can add to the `correct vault namespace annotations is set when global.secretsBackend.vault.vaultNamespace is set and agentAnnotations are also set` test?

Reply:
@absolutelightning if I set an unrelated annotation (not namespace) via `agentAnnotations`, wouldn't we still expect setting `vaultNamespace` to add the namespace annotation? My read of this is that we have made the two config values mutually exclusive, even if `agentAnnotations` does not contain `vaultNamespace`.

Reply:
According to me we will not know if namespace annotation is present or not in agent annotations added by user. So Ideally we would check if namespace annotation is present or not in agent annotations and then add namespace annotation in case user sets vaultNamespace. But I am not sure if it is possible.

Reply:
yes.

Reply:
> > Ideally we would check if namespace annotation is present or not in agent annotations and then add namespace annotation in case user sets vaultNamespace.
> 
> Here is my thinking of what I expect in the following cases:
> 
> 1. `vaultNamespace` set and no namespace annotation in `agentAnnotations` - add the agent annotations to the existing annotations the user provides
> 2. No `vaultNamespace` set and `agentAnnotations` namespace set - Just use the agentAnntotations that have been set which includes that namespace annotation which the user manually set.
> 3. `vaultNamespace` set and namespace annotation in `agentAnnotations` set - Leave annotations as is. We'd bet setting the ConnectCA in one namespace and set agent annotations for the vault agent with another for KV storage.
> 4. `vaultNamespace` not set and no `agentAnnotations` set - Do nothing.
> 
> I think I covered all permutations?

All cases are handled in bats test @zalimeni